Source/items.cpp

```diff
@@ -3791,292 +3791,226 @@ void DoOil(Player &player, int cii)
 	case IPL_MULT_ARROWS:
-		strcpy(tempstr, _("multiple arrows per shot"));
-		break;
+		return _("multiple arrows per shot");
 	case IPL_FIRE_ARROWS:
-		if (x->_iFMinDam == x->_iFMaxDam)
-			strcpy(tempstr, fmt::format(_("fire arrows damage: {:d}"), x->_iFMinDam).c_str());
+		if (item._iFMinDam == item._iFMaxDam)
+			return fmt::format(_("fire arrows damage: {:d}"), item._iFMinDam);
 		else
-			strcpy(tempstr, fmt::format(_("fire arrows damage: {:d}-{:d}"), x->_iFMinDam, x->_iFMaxDam).c_str());
-		break;
+			return fmt::format(_("fire arrows damage: {:d}-{:d}"), item._iFMinDam, item._iFMaxDam);
 	case IPL_LIGHT_ARROWS:
-		if (x->_iLMinDam == x->_iLMaxDam)
-			strcpy(tempstr, fmt::format(_("lightning arrows damage {:d}"), x->_iLMinDam).c_str());
+		if (item._iLMinDam == item._iLMaxDam)
+			return fmt::format(_("lightning arrows damage {:d}"), item._iLMinDam);
 		else
-			strcpy(tempstr, fmt::format(_("lightning arrows damage {:d}-{:d}"), x->_iLMinDam, x->_iLMaxDam).c_str());
-		break;
+			return fmt::format(_("lightning arrows damage {:d}-{:d}"), item._iLMinDam, item._iLMaxDam);
 	case IPL_FIREBALL:
-		if (x->_iFMinDam == x->_iFMaxDam)
-			strcpy(tempstr, fmt::format(_("fireball damage: {:d}"), x->_iFMinDam).c_str());
+		if (item._iFMinDam == item._iFMaxDam)
+			return fmt::format(_("fireball damage: {:d}"), item._iFMinDam);
 		else
-			strcpy(tempstr, fmt::format(_("fireball damage: {:d}-{:d}"), x->_iFMinDam, x->_iFMaxDam).c_str());
-		break;
+			return fmt::format(_("fireball damage: {:d}-{:d}"), item._iFMinDam, item._iFMaxDam);
 	case IPL_THORNS:
-		strcpy(tempstr, _("attacker takes 1-3 damage"));
-		break;
+		return _("attacker takes 1-3 damage");
 	case IPL_NOMANA:
-		strcpy(tempstr, _("user loses all mana"));
-		break;
+		return _("user loses all mana");
 	case IPL_NOHEALPLR:
-		strcpy(tempstr, _("you can't heal"));
-		break;
+		return _("you can't heal");
 	case IPL_ABSHALFTRAP:
-		strcpy(tempstr, _("absorbs half of trap damage"));
-		break;
+		return _("absorbs half of trap damage");
 	case IPL_KNOCKBACK:
-		strcpy(tempstr, _("knocks target back"));
-		break;
+		return _("knocks target back");
 	case IPL_3XDAMVDEM:
-		/*xgettext:no-c-format*/ strcpy(tempstr, _("+200% damage vs. demons"));
-		break;
+		return _("+200% damage vs. demons");
 	case IPL_ALLRESZERO:
-		strcpy(tempstr, _("All Resistance equals 0"));
-		break;
+		return _("All Resistance equals 0");
 	case IPL_NOHEALMON:
-		strcpy(tempstr, _("hit monster doesn't heal"));
-		break;
+		return _("hit monster doesn't heal");
 	case IPL_STEALMANA:
-		if ((x->_iFlags & ISPL_STEALMANA_3) != 0)
-			/*xgettext:no-c-format*/ strcpy(tempstr, _("hit steals 3% mana"));
-		if ((x->_iFlags & ISPL_STEALMANA_5) != 0)
-			/*xgettext:no-c-format*/ strcpy(tempstr, _("hit steals 5% mana"));
-		break;
+		if ((item._iFlags & ISPL_STEALMANA_3) != 0)
+			return _("hit steals 3% mana");
+		if ((item._iFlags & ISPL_STEALMANA_5) != 0)
+			return _("hit steals 5% mana");
+		return _("Another ability (NW)");

```

Comment:
Previously tempstr was left with the result of the last successful call. Even though in practice this case should never be hit could we use either `""` or a more descriptive string for these cases?
```suggestion
		return _("hit steals mana");
```

If you wanted to put a number on it "hit steals 0% mana" is accurate 😂

Reply:
I think I'll go with `""`. No need for adding new strings to be translated if they are never used :)
```suggestion
		return "";
```